homework/index.js

```diff
@@ -29,13 +29,47 @@
     return elem;
   }
 
+  // 1- create function to add row
+  function createRow(table, label, content) {
+    const row = createAndAppend('tr', table);
+    createAndAppend('th', row, { text: label });
+    createAndAppend('td', row, { text: content });
+    return row;
+  }
+
+  // 2-  append details of repositories to list
   function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+    const li = createAndAppend('li', ul, { class: 'item' });
+    const table = createAndAppend('table', li);
+    // 2-1 add repository name
+    const repoName = createRow(table, 'Repository:', '');
+    createAndAppend('a', repoName.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+      target: '_blank',
+    });
+    // 2-2 add description
+    if (repo.description) {
+      createRow(table, 'Description:', repo.description);
+    } else {
+      createRow(table, 'Description:', 'N/A');
+    }
```

Comment:
Final tip: no need for an `if` statement.

```js
// 2-2 add description
createRow(table, 'Description:', repo.description || 'N/A');
```

The value of the expression ` repo.description || 'N/A'` is the value of `repo.description` if that value is truthy, or `N/A` if it is falsy (which includes `null`. `undefined` and the empty string `''`).